lib/linguist/heuristics.yml

```
@@ -107,7 +107,7 @@ disambiguations:
 - extensions: ['.cls']
   rules:
   - language: TeX
-    pattern: '\\\w+{'
+    pattern: '^\s*\\ProvidesClass{'
```

Could you include `NeedsTeXFormat` as well, please? I probably should've been clearer in my [earlier comment](https://github.com/github/linguist/issues/5219#issuecomment-782545817).

```suggestion
    pattern: '^\s*\\(?:NeedsTeXFormat|ProvidesClass){'
```

I don't know TeX well enough to know if `ProvidesClass` can or can't be omitted or not, and `NeedsTeXFormat` has virtually zero risk of appearing in a `.cls` file that isn't TeX.

---

Yeah this is a fair point, it generally seemed that NeedsTeXFormat could be ommitted so wouldn't be a requirement.

But yes, not really a problem having it.